### DISCUSSION TITLE
Applying Y.XmlElement attributes to node?

### DISCUSSION THREAD

**Discussion by jloganolson** [ueberdosis/tiptap#1795] [General Questions] [answered]
So I've figured out how to roll my own Y.XmlFragments (so I can generate TipTap docs without any editors open), but I've hit a roadblock trying to include node attributes. Has any one else managed to do this?

This does not work:
```
 const gen = new Y.XmlElement('newdoc');
 gen.setAttribute('data-label', 'docs label!')
```

The node generated from that element still has the default data-label:
`div data-label="Generated"`

Any ideas?

**Comment by hanspagel** [comment]
Are you using hocuspocus? It has a JSON to YDoc Transformer.

**Reply by jloganolson** [reply]
Not yet - still local. That being said, I just looked at the code in hocuspocus and it's identical! Have you managed to pass attributes from that json/fragment to a node view renderer?

**Reply by hanspagel** [reply]
Hmm ..  that shouldn’t differ from a normal node.

**Reply by jloganolson** [reply]
Updated the top discussion to reflect that this applies to all nodes

**Reply by jloganolson** [reply]
One more update - style works as an attiribute but not data-label:
```
 const gen = new Y.XmlElement('newdoc');
 gen.setAttribute('data-label', 'yahoo!') // doesn't pass to node
 gen.setAttribute('style', 'red') // passes to node
```

Is there only some attributes that will pass? Any that I can use to pass an arbitrary string?

**Comment by jloganolson** [comment, marked as answer]
Resolved: Was using the html attribute name rather than the attributes key - confirmed that this works in both regular and node views when using the appropriate attribute names
